Add Erwerbsminderungsrente (#577)
iza-institute-of-labor-economics/gettsim: docs/gettsim_objects/input_variables.md

Number of own children below the age of 25 (incl. not in hh)

Type: int

## `erwerbsgemindert`

> Dummy: Beeing able to work less than 3 hours on the general labor market per day

Reviewer: ```suggestion
Unable to provide more than 3 hours of market labor per day.
```

Type: bool

## `teilw_erwerbsm_rente`

> Dummy: Beeing able to work between 3 and 6 hours on the general labor market per day

Reviewer: ```suggestion
Able to provide at least 3 but no more than 6 hours of market labor per day.
```

Type: bool

> ## `erwerbsm_rente_mit_63`

Reviewer: This is opaque to me. The only explanation is the line below; I don't know what the conditions for this would be in the first place?

Also, please use the official term rather than the colloquial one.

Author: As far as I know, there is no technical term for this. This is an exceptional case in the calculation of the access factor. I have added a small explanatory text. Is that sufficient?

The legal foundation is
§ 77 Abs. 4  SGB VI
> (4) Bei Renten wegen verminderter Erwerbsfähigkeit und bei Hinterbliebenenrenten, deren Berechnung 40 Jahre mit den in § 51 Abs. 3a und 4 und mit den in § 52 Abs. 2 genannten Zeiten zugrunde liegen, sind die Absätze 2 und 3 mit der Maßgabe anzuwenden, dass an die Stelle der Vollendung des 65. Lebensjahres die Vollendung des 63. Lebensjahres und an die Stelle der Vollendung des 62. Lebensjahres die Vollendung des 60. Lebensjahres tritt.

Reviewer: How do we call the corresponding parameter in regular pensions then?

Author: I think it is very similar to the rules for `langjährig und besonders langjährig Versicherte`in the regular pension.
But I have not yet seen this term beeing used for the disability pension. But I think we can adopt the name, to at least make it consistent in GETTSIM.

Reviewer: Oh, lovely, and it is yet another number (40 instead of 35/45), right? 

But yeah, I'd think `besond_langj` would be best. Seems most similar in spirit. Also note that the text also has the 62->60 mapping, hence the 63 doesn't quite capture it. Btw, is that relevant for us?

Author: Ok I will rename that parameter accordingly.
The jump from 62 to 60 refers to the maximum deduction that is specified elsewhere in the code.
The limit is either 40 years or, in special cases, 35 years. 
> - abschlagsfreier Beginn der Rente mit Vollendung des 63. Lebensjahres, wenn 40 Jahre Wartezeit nach § 51 Absatz 3a SGB VI vorliegen oder
> - abschlagsfreier Beginn nach den Tabellenwerten des § 264d SGB VI, wenn mindestens 35 Jahre Wartezeit nach § 51 Absatz 3a SGB VI vorliegen- diese Regelung gilt aber nur noch für EM-Rente die bis zum 31.12.2023 beginnen (Auslaufregelung).

Reviewer: I just don't overlook whether these two points are part of the code right now? If not, it seems like we'll want to include them in this PR, too?

Author: We discussed this and decided that it was too complicated to get these time variables from the data. It needs some kind of approximation which is better done by the user and not by Gettsim.

Reviewer: The user would have to do the approximation indeed (unless she had data from the DRV), but GETTSIM needs to know the rule. Same as everywhere?

Reviewer 2: Hi, I assumed the variable `erwerbsm_langj_vers` already indicates this approximation/exception set by the user? 

If we want GETTSIM to process different types of time variables/time thresholds that the German pension system recognizes, I think it would make sense to do this in a separate PR, as this will affect other types of pensions as well.

Reviewer: Oh shoot, good that you are writing this. I completely missed that. 

This is exactly how, e.g., the `ges_rente_vorauss_besond_langj` works, no? We should be symmetric here, shouldn't be hard -- the whole machinery is in place already.

I am terribly sorry I did not realise this before.

Reviewer 2: Thanks! This is my understanding as well. What exactly do you mean by symmetry: the name? In that case I would suggest renaming the variable once more to `erwerbsm_vorauss_besond_langj` in line with `ges_rente_vorauss_besond_langj`. Not sure if the `besond` is necessary, but I agree it is more in line with the spirit of the rules for old age pensions.

Reviewer: What I mean by symmetry is (correct me if I am wrong, did not check, might misremeber):

For old age pensions we expect eligibility conditions (contribution periods, age) and calculate eligibility.

We should do the same here (contribution periods, age, disability status), rather than expecting eligibility as an input.

Note that given the DAG-structure, you can trivially pass eligibility. It just should not be the primitive, I think.

Number of own children below the age of 25 (incl. not in hh)

Type: int

> ## `erwerbsgemindert`

Reviewer: ```suggestion
## `voll_erwerbsgemindert`
```
I think that will make things clearer, but feel free to leave it as is if I am misunderstanding something.

Dummy: Unable to provide more than 3 hours of market labor per day.

Type: bool

> ## `teilw_erwerbsm_rente`

Reviewer: ```suggestion
## `teilw_erwerbsgemindert`
```

Should be symmetric to above, I think.
